### ISSUE TITLE
Separate config for nether and overworld trees/nether wart blocks breaking adjustment

### ISSUE THREAD

**Issue by goodideagiver** [RakambdaOrg/FallingTree#126]
When using `shift down` option on nether trees there should be a separate option for nether trees leaves to be destroyed instantly, because otherwise breaking of the trees using this method is very slow since every wart/leaf block is broken with your axe one by one.

Alternative 1: Separate breaking speed option for wart blocks

Alternative 2: Decay wart blocks like overworld trees leaves

**Comment by Rakambda** [comment]
Yeah good point.

1 - Will probably do this approach. Since the warts are meant to be broken with a hoe it makes sense that with the axe it's slower.

2 - I don't really like this approach as the warts are not decayable but meant to be broken by you (the big difference is that you damage your tool with warts while you don't with decaying leaves)

3 - What if there's a true/false option that allows "Instant wart breaking" and that basically destroys all the warts with only one cut while in shift down ? Does it sound interesting (ie: First cut, you break all warts, second and the followings it's the trunk (and ofc the tool gets damaged for all blocks))

**Comment by goodideagiver** [comment]
1 - To be precise I thought of wart blocks faster breaking option because now it takes forever to see first nether log when using`shift down `

3 - that sounds cool, but maximum distance from trunk needs to be specified. I realized that using `shift down` tends to suck adjacent trees wart blocks

**Comment by Rakambda** [comment]
For the first point do you have an example of when it eats up the warts of another tree? 

It currently keeps only warts that are maximum 4 blocks (max from the generation) away from from the hit position (only on the x/z plane). Maybe a check should be added on the y value? Like when it happens it's because trees are "stacked" and it destroys a bit column of warts?

**Comment by Rakambda** [comment]
I thought a bit about it but in general wouldn't it makes more sense while you're in shift down mode to simply deactivate the "break warts" flag and simply break them manually with the appropriate tool?

**Comment by goodideagiver** [comment]
> I thought a bit about it but in general wouldn't it makes more sense while you're in shift down mode to simply deactivate the "break warts" flag and simply break them manually with the appropriate tool?

This option would also be nice to have. Personally I find breaking wart blocks extremely boring and that's why I suggested above solutions.

Another interesting option would be shift down mode for a hoe to remove wart blocks 🙂

**Comment by Rakambda** [comment]
So I went on and implemented the option :
> 3 - What if there's a true/false option that allows "Instant wart breaking" and that basically destroys all the warts with only one cut while in shift down ? Does it sound interesting (ie: First cut, you break all warts, second and the followings it's the trunk (and ofc the tool gets damaged for all blocks))

And it looks something like this (with the option turned on):
https://youtu.be/QQ5eBxDSe9Y

The warts detection isn't the best for sure when trees are close together but I can't do any better than this. Trees are checked in a 9*9 are centered on the trunk (this is the max size of a nether tree, see at 01:07).
Though I think it could answer your need as warts should now be broken faster even if there's a bit of potential collateral damage if trees are close.

I may consider `Another interesting option would be shift down mode for a hoe to remove wart blocks` as a more elegant solution but not for now. This would require much more work on it.

**Comment by goodideagiver** [comment]
That looks great. I hoped that for `Another interesting option would be shift down mode for a hoe to remove wart blocks ` option you could just reuse code for logs and make axe ignore warts and just chop down the wood.

**Comment by Rakambda** [comment]
I tried to do different things to make the config clearer and maybe do the shift down with hoes but for now it's either breaking everything or get an even messier config x).

So for now i'll give you the jar if you wanna try for yourself and I'll release it in a few days like this I think.
Then eventually when I have a better idea on how to do this without breaking everything then I'll consider doing the shift down with hoes.
It's 1.17 fabric only though, idk what version you use. https://www.dropbox.com/t/dyb5206GDQEAnwdh

**Comment by goodideagiver** [comment]
I tested the jar and the new functionality for breaking warts works nice! Hovewer I found out that It's easy to loose your axe because I think there is no protection if axe has too little durability to eat the warts (there is also no sound if the axe breaks). 

I'm looking forward to shift down for wart blocks being implemented in the future :)

*It happened only once but I think there was a situation where after wart blocks being broken two pillars of wart logs standing 1 block apart were being chopped (shift down) even when there was air gap between them. If you want I can run more tests to confirm the problem.

**Comment by Rakambda** [comment]
With preserve tools : https://www.dropbox.com/t/nmA1cTGEUnXbHsOa

> *It happened only once but I think there was a situation where after wart blocks being broken two pillars of wart logs standing 1 block apart were being chopped (shift down) even when there was air gap between them. If you want I can run more tests to confirm the problem.

If you manage to reproduce it and have a clearer idea on the situation when this happens I can look into it. Maybe this can be done in a separate issue though? Like that if this version seems fine for you I can already release that.
